trie/sync.go

```diff
@@ -63,12 +65,18 @@ type SyncPath [][]byte
 
 // NewSyncPath converts an expanded trie path from nibble form into a compact
 // version that can be sent over the network.
-func NewSyncPath(path []byte) SyncPath {
+func NewSyncPath(path []byte, isZk bool) SyncPath {
 	// If the hash is from the account trie, append a single item, if it
 	// is from a storage trie, append a tuple. Note, the length 64 is
 	// clashing between account leaf and storage root. It's fine though
 	// because having a trie node at 64 depth means a hash collision was
 	// found and we're long dead.
+	if isZk {
+		if len(path) < 256 {
+			return SyncPath{hexToCompact(path)}
+		}
+		return SyncPath{path[:256], hexToCompact(path[256:])}
```

Comment:
The latter part (`hexToCompact(path[256:]`)is for storage?

Reply:
yes